src/js/control-bar/text-track-controls/chapters-button.js

```diff
@@ -124,17 +133,26 @@ class ChaptersButton extends TextTrackButton {
 
         menu.addChild(mi);
       }
-
-      this.addChild(menu);
     }
 
-    if (this.items.length > 0) {
+    if (items.length > 0) {
       this.show();
     }
-
+    // Assigning the value of items back to this.items for next iteration
+    this.items = items;
     return menu;
   }
-
+  // Overriding the update button because duplicate chapters get created each time when
+  // we try to put the chapter title into the DOM
+
+  update() {
+      if(this.children_.length <= 1) {
+        this.firstCall = true;
+        super.update();
```

Comment:
This can still be called if a non-chapters track is added. We should make sure that it only does anything when chapters are added.
The `update` method is an event handler.